iree/hal/local/loaders/BUILD

```diff
@@ -94,3 +94,33 @@ endif()
 """,
     inline = True,
 )
+
+# TODO(scotttodd): add to Bazel when the WASM driver builds with it
+iree_cmake_extra_content(
+    content = """
+if(${IREE_HAL_DRIVER_WASM})
+
+iree_cc_library(
+  NAME
+    wasm_module_loader
+  HDRS
+    "wasm_module_loader.h"
+  SRCS
+    "wasm_module_loader.c"
+  DEPS
+    iree::base::api
+    iree::base::flatcc
+    iree::base::tracing
+    iree::hal::api
+    iree::hal::local
+    iree::schemas::dylib_executable_def_c_fbs
+    wasm_micro_runtime_vmlib
```

Comment:
A few build errors on Android with this: [logs](https://source.cloud.google.com/results/invocations/bc6c1924-7f97-4766-b7a8-cc69dd00108f/targets/iree%2Fgcp_ubuntu%2Fcmake%2Fandroid%2Farm64-v8a%2Fpresubmit/log)

```
../third_party/wasm-micro-runtime/core/iwasm/common/arch/invokeNative_em64.s:19:10: error: unknown token in expression
    push %rbp
         ^
../third_party/wasm-micro-runtime/core/iwasm/common/arch/invokeNative_em64.s:19:10: error: invalid operand
    push %rbp
         ^
```

```
../third_party/wasm-micro-runtime/core/shared/platform/common/posix/posix_memmap.c:40:22: error: use of undeclared identifier 'MAP_32BIT'
        map_flags |= MAP_32BIT;
                     ^
```